src/main/java/hudson/remoting/Launcher.java

```diff
@@ -226,9 +263,46 @@ public static void main(String... args) throws Exception {
 
     @edu.umd.cs.findbugs.annotations.SuppressWarnings("DM_DEFAULT_ENCODING")    // log file, just like console output, should be in platform default encoding
     public void run() throws Exception {
-        if (slaveLog!=null) {
+
+        // Create and verify working directory if required
+        @CheckForNull Path internalDirPath = workDir != null ? workDir.toPath() : null;
+        if (internalDirPath == null) {
+            System.err.println("WARNING: Agent working directory is not specified. Some functionality introduced in Remoting 3 may be disabled");
+        } else {
+            if (workDir.exists()) {
+                if (!workDir.isDirectory()) {
+                    throw new IOException("The specified agent working directory path points to a non-directory file");
+                }
+                if (!workDir.canWrite() || !workDir.canRead() || !workDir.canExecute()) {
```

Comment:
Should/couldn't it just (try to) create it? Like for `.jenkins` will just be created if absent?